lib/logproto/logproto-proxied-text-server.c

```diff
@@ -189,6 +189,41 @@ _log_proto_proxied_text_server_add_aux_data(LogProtoProxiedTextServer *self, Log
   return;
 }
 
+static inline LogProtoStatus
+_fetch_into_proxy_buffer(LogProtoProxiedTextServer *self, gsize *hdr_len)
+{
+  for(gint i = 0; i < PROXY_PROTO_HDR_MAX_LEN_RFC; i++)
+    {
+      gssize rc = log_transport_read(self->super.super.super.transport, &(self->v1_proxy_header_buff[i]), sizeof(gchar),
+                                     NULL);
+      if(rc > 0)
+        {
+          self->v1_proxy_header_buff[i+1] = '\0';
+          *hdr_len = i+1;
+          if (self->v1_proxy_header_buff[i] == '\n')
+            {
+              return LPS_SUCCESS;
+            }
+        }
+      else if(rc == 0)
+        {
+          return LPS_AGAIN;
```

Comment:
This is an EOF situation, which usually means `LPS_EOF`.

LPS_AGAIN can be detected with `rc < 0 && errno == EAGAIN`